blueflood-core/src/main/java/com/rackspacecloud/blueflood/io/serializers/metrics/BaseRollupSerDes.java

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright 2016 Rackspace
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package com.rackspacecloud.blueflood.io.serializers.metrics;
+
+import com.codahale.metrics.Histogram;
+import com.google.protobuf.CodedInputStream;
+import com.google.protobuf.CodedOutputStream;
+import com.rackspacecloud.blueflood.exceptions.SerializationException;
+import com.rackspacecloud.blueflood.io.Constants;
+import com.rackspacecloud.blueflood.types.BaseRollup;
+import com.rackspacecloud.blueflood.utils.Metrics;
+
+import java.io.IOException;
+
+/**
+ * The serialization and deserialization methods for sub-metrics used by BasicMetricRollup and GaugeMetricRollup.
+ */
+public class BaseRollupSerDes extends AbstractSerDes {
```

Comment:
Minor comment: I wonder if we should pick a more meaningful name than ```BaseRollupSerDes```? Maybe ```SubStatRollupSerDes``` ? Or ```AvgVarMinMaxRollupSerDes``` ?